src/status_im/common/standard_authentication/events.cljs

```diff
@@ -94,6 +108,42 @@
         :button-icon-left    auth-button-icon-left
         :button-label        auth-button-label}])))
 
+(defn get-signature-map
+  [tx-hash signature]
+  {tx-hash {:r (subs signature 0 64)
+            :s (subs signature 64 128)
+            :v (str (js/parseInt (subs signature 128 130) 16))}})
+
+(defn authorize-with-keycard
+  [{:keys [db]} [tx-hash]]
+  (let [current-address (get-in db [:wallet :current-viewing-account-address])
+        path            (get-in db [:wallet :accounts current-address :path])
+        key-uid         (get-in db [:profile/profile :key-uid])]
+    {:fx [[:dispatch
+           [:show-bottom-sheet
+            {:on-close #(rf/dispatch [:standard-auth/reset-login-password])
+             :content  (fn []
+                         [keycard.pin/auth
+                          {:on-complete
+                           (fn [pin-text]
+                             (rf/dispatch
+                              [:keycard/read-card
+                               {:key-uid    key-uid
+                                :on-read-fx [[:effects.keycard/sign
+                                              {:pin pin-text
+                                               :path path
+                                               :hash (utils.address/naked-address tx-hash)
+                                               :on-success
+                                               #(do
+                                                  (rf/dispatch [:keycard/hide-connection-sheet])
+                                                  (rf/dispatch
+                                                   [:wallet/proceed-with-transactions-signatures
+                                                    (get-signature-map tx-hash %)]))
+                                               :on-failure #(rf/dispatch
+                                                             [:keycard/on-action-with-pin-error
+                                                              %])}]]}]))}])}]]]}))
+(rf/reg-event-fx :standard-auth/authorize-with-keycard authorize-with-keycard)
```

Comment:
Since we use this event only to sign the transaction, it's better we rename it to what it does. WDYT?

In future, we will need to authorize the user with keycard for several actions such as creating a new wallet account, getting a new sync code for pairing, ...etc

Reply:
yes good point, probably better to move auth :on-complete outside as parameter so authorize-with-keycard will stay general